pkg/pvc/pvc_protected_entity.go

```diff
@@ -201,14 +200,34 @@ func (this PVCProtectedEntity) DeleteSnapshot(ctx context.Context, snapshotToDel
 			return false, errors.Wrapf(err, "Could not retrieve snapshot configmap %s for %s", snapConfigMapName,
 				this.id.String())
 		}
+		this.logger.Infof("PVCProtectedEntity: No snapshot config map avaiable with name: %s", snapConfigMapName)
 		// No configmap so no snapshots
 		return false, nil
 	}
 
-	_, snapPVCInfoExists := snapConfigMap.BinaryData[snapshotToDelete.String()]
+	components, err := this.GetComponents(ctx)
+	if err != nil {
+		return false, errors.Wrap(err, "Could not retrieve components")
+	}
+	if len(components) != 1 {
+		return false, errors.New(fmt.Sprintf("Expected 1 component, %s has %d", this.id.String(), len(components)))
+	}
+	this.logger.Infof("PVCProtectedEntity: Retrieved components for the pvc protected entity id: %s", components[0].GetID().String())
+	componentSnapshotID := components[0].GetID().GetSnapshotID()
+	this.logger.Infof("snapshotToDelete: %s  componentSnapshotID: %s", snapshotToDelete.String(), componentSnapshotID.String())
+	deleteSnapStatus := false
+	this.logger.Infof("PVCProtectedEntity: Triggering DeleteSnapshot with component snapshotID: %s for the component pe-id: %s", componentSnapshotID.String(), components[0].GetID().String())
+	_, err = components[0].DeleteSnapshot(ctx, componentSnapshotID, params)
+	if err != nil {
+		this.logger.Warnf("Subcomponent peid %s delete snapshot failed error: %v", components[0].GetID(), err)
+	} else {
+		deleteSnapStatus = true
+		this.logger.Infof("PVCProtectedEntity: Completed DeleteSnapshot with snapshotID: %s for the component: %s", snapshotToDelete.String(), components[0].GetID().String())
+	}
+	_, snapPVCInfoExists := snapConfigMap.BinaryData[componentSnapshotID.String()]
 	if snapPVCInfoExists {
-		delete(snapConfigMap.BinaryData, snapshotToDelete.String())
-		delete(snapConfigMap.BinaryData, "peinfo-"+snapshotToDelete.String())
+		delete(snapConfigMap.BinaryData, componentSnapshotID.String())
```

Comment:
This should really be the PVC snapshot ID here.  The list in the snapConfigMap is the list of all snapshots for the PVC PE, not the components

Reply:
moved this under a false block as spoken offline.